pkg/collector/corechecks/snmp/internal/checkconfig/config_metric.go

```diff
@@ -19,7 +19,7 @@ func BuildMetricTagsFromValue(metricTag *profiledefinition.MetricTagConfig, valu
 	var tags []string
 	if metricTag.Tag != "" {
 		if len(metricTag.Mapping) > 0 {
-			mappedValue, err := GetMappedValue(value, metricTag.Mapping)
+			mappedValue, err := GetMappedValue(value, metricTag.Mapping.ToMap())
```

Comment:
Note: if needed we can cache the map conversion on `metricTag` to avoid having to re-compute, but the operation should be quite fast, so we likely don't need to optimise for now.